tests/unit/moduleapi/internalsecret.tcl

```diff
@@ -0,0 +1,275 @@
+        set reply [r internalauth.getinternalsecret]
+        assert_equal {OK} [r auth "internal connection" $reply]
+
+        # Call an internal writing command
+        assert_equal {OK} [r internalauth.internal_rmcall_replicated set x 5]
+
+        # Reload the server from the AOF
+        r debug loadaof
+
+        # Check if the internal command was executed successfully
+        assert_equal {5} [r get x]
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set r [srv 0 client]
+
+    test {Internal commands are not allowed from scripts} {
+        # Internal commands are not allowed from scripts
+        assert_error {*not allowed from script*} {r eval {redis.call('internalauth.internalcommand')} 0}
+
+        # Even after authenticating as an internal connection
+        set reply [r internalauth.getinternalsecret]
+        assert_equal {OK} [r auth "internal connection" $reply]
+        assert_error {*not allowed from script*} {r eval {redis.call('internalauth.internalcommand')} 0}
+
+        # Internal commands ARE shown in monitor output
+        set rd [redis_deferring_client]
+        $rd monitor
+        $rd read ; # Discard the OK
+        catch {r eval {redis.call('internalauth.internalcommand')} 0} err
+        assert_match "*not allowed from script*" $err
+        assert_match {*eval*internalauth.internalcommand*} [$rd read]
+        # No following log, since the command failed.
+        $rd close
+    }
+}
+
+start_cluster 1 1 [list config_lines $modules] {
+    set master [srv 0 client]
+    set slave [srv -1 client]
+
+    test {Setup master} {
+        # Authenticate as an internal connection
+        set reply [$master internalauth.getinternalsecret]
+        assert_equal {OK} [$master auth "internal connection" $reply]
+    }
+
+    test {Slaves successfully execute internal commands from replication link} {
+        assert {[s -1 role] eq {slave}}
+        wait_for_condition 1000 50 {
+            [s -1 master_link_status] eq {up}
+        } else {
+            fail "Master link status is not up"
+        }
+
+        # Execute internal command in master, that will set `x` to `5`.
+        assert_equal {OK} [$master internalauth.internal_rmcall_replicated set x 5]
+
+        # Wait for replica to have the key
+        $slave readonly
+        wait_for_condition 1000 50 {
+            [$slave exists x] eq "1"
+        } else {
+            fail "Test key was not replicated"
+        }
+
+        # See that the slave has the same value for `x`.
+        assert_equal {5} [$slave get x]
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set master [srv 0 client]
+
+    test {Internal commands are not reported in the monitor output for non-internal connections} {
+        # Execute an internal command
+        set rd [redis_deferring_client]
+        $rd monitor
+        $rd read ; # Discard the OK
+        assert_error {*unknown command*} {r internalauth.internalcommand}
```

Comment:
Let authenticate as internal connection and make sure that the internal command are not shown even when the command succeed.

Reply:
So the command still fails here no?

Reply:
Ohh ok it is down bellow :+1: